lib/src/default_index/revset_engine.rs

```diff
@@ -1204,35 +1204,35 @@ fn build_predicate_fn(
             })
         }
         RevsetFilterPredicate::Description(pattern) => {
-            let pattern = pattern.clone();
+            let matcher = Rc::new(pattern.to_matcher());
```

Comment:
nit: maybe these also should be `Arc`'s since the code is used in ERSC's backend.